File: src/main/java/com/williamfiset/algorithms/math/StandardDeviation.java
package com.williamfiset.algorithms.math;

public class StandardDeviation {

    public static double SD(double arr[]) {
        double sum = 0, StandardDeviation = 0;
        int length = arr.length;
        for (double num : arr) {
            sum += num;
        }
        double mean = sum / length;
        for (double num : arr) {
            StandardDeviation = StandardDeviation + Math.pow(num - mean, 2);
        }
        return Math.sqrt((StandardDeviation / length));
    }
}

File: standarddeviation.cpp
#include<iostream>
#include<vector>
#include <cmath>
using namespace std;

int main() {
    
    vector<float>s1;//Vector create for storing value dynamically.
   float sum = 0.0, mean, variance = 0.0, stddev;
   int n;
   cin>>n;
   int i;
   for( i= 0; i < n; i++)
   {
      float value;
      cin>>value;
      s1.push_back(value);
   }
   for(i = 0; i < n; i++)
   {
       sum=sum+s1[i];
   }
   
      
   mean = sum/5;

   for(i = 0; i < n; i++)
   variance += pow(s1[i] - mean, 2);

   variance=variance/5;//For variance.
    
   stddev = sqrt(variance);//Formula for calculate standard deviation.
    
   cout<<"The give stored values:- ";
    
   for(i = 0; i < n; i++)
      {
          cout<<s1[i]<<" ";
      }
    
      cout<<endl;

      cout<<"Standered deviation:-"<<stddev; //Output.   
}
